common/gatewayconfig.cpp

```diff
@@ -77,8 +88,8 @@ bool GatewayConfiguration::append(const std::string &id, json_t *sourceArray)
         }
         json_decref(backupArray);
     } else {
-        m_configMap[id] = sourceArray;
-        json_incref(sourceArray);
+        m_configMap[id] = json_deep_copy(sourceArray);
+        // we do not need to decref the sourceArray since we did a deep copy
```

Comment:
It's a comment for why we don't do anything with the source that we got from the caller. The reader be all like "hey, why no code here?", and we be like "cause we copy deep".